common/src/main/java/io/homeassistant/companion/android/common/sensors/DisplaySensorManager.kt

```diff
@@ -98,4 +149,104 @@ class DisplaySensorManager : SensorManager {
+        if (!isEnabled(context, screenOrientation)) {
+            return
+        }
+
+        @Suppress("DEPRECATION")
+        val orientation = when (context.resources.configuration.orientation) {
+            Configuration.ORIENTATION_PORTRAIT -> "portrait"
+            Configuration.ORIENTATION_LANDSCAPE -> "landscape"
+            Configuration.ORIENTATION_SQUARE -> "square"
+            Configuration.ORIENTATION_UNDEFINED -> STATE_UNKNOWN
+            else -> STATE_UNKNOWN
+        }
+
+        val icon = when (orientation) {
+            "portrait" -> "mdi:phone-rotate-portrait"
+            "landscape" -> "mdi:phone-rotate-landscape"
+            "square" -> "mdi:crop-square"
+            else -> screenOrientation.statelessIcon
+        }
+
+        onSensorUpdated(
+            context,
+            screenOrientation,
+            orientation,
+            icon,
+            mapOf()
+        )
+    }
+
+    private fun updateScreenRotation(context: Context) {
+        if (!isEnabled(context, screenRotation)) {
+            return
+        }
+
+        val wm = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+
+        @Suppress("DEPRECATION")
+        val display = when (wm.defaultDisplay.rotation) {
+            Surface.ROTATION_0 -> "0°"
+            Surface.ROTATION_90 -> "90°"
+            Surface.ROTATION_180 -> "180°"
+            Surface.ROTATION_270 -> "270°"
+            else -> STATE_UNKNOWN
+        }
+
+        onSensorUpdated(
+            context,
+            screenRotation,
+            display,
+            screenRotation.statelessIcon,
+            mapOf()
+        )
+    }
+
+    private fun updateIsFaceDown(context: Context) {
+        if (!isEnabled(context, isFaceDown)) {
+            return
+        }
+        val now = System.currentTimeMillis()
+        if (listenerLastRegistered + SensorManager.SENSOR_LISTENER_TIMEOUT < now && isListenerRegistered) {
+            Log.d(TAG, "Re-registering listener as it appears to be stuck")
+            mySensorManager.unregisterListener(this)
+            isListenerRegistered = false
+        }
+        mySensorManager = latestContext.getSystemService()!!
+
+        val accelerometerSensor = mySensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)
+        if (accelerometerSensor != null && !isListenerRegistered) {
+            mySensorManager.registerListener(
+                this,
+                accelerometerSensor,
+                SENSOR_DELAY_NORMAL
+            )
+            Log.d(TAG, "Accelerometer sensor listener registered")
+            isListenerRegistered = true
+            listenerLastRegistered = now.toInt()
+        }
+    }
+
+    override fun onSensorChanged(event: SensorEvent?) {
+        if (event?.sensor?.type == Sensor.TYPE_ACCELEROMETER) {
+            onSensorUpdated(
+                latestContext,
+                isFaceDown,
+                event.values[2] < -9,

```

Comment:
So I think this is from [here](https://developer.android.com/reference/android/hardware/SensorEvent#values) and inverted for the flat on the table example? I think it might be helpful to add a note here why this value is set 😅